.gitignore

```diff
@@ -3,4 +3,9 @@
 /build/*
 /.gradle/*
 /ui/build/*
-/api/build/*
\ No newline at end of file
+/api/build/*
+node_modules
+npm-debug.log
+*.map
+*.js
```

Comment:
Shouldn't we instead exclude the directory that contains generated Javascript?

I don't know if we ever intend to check Javascript directly into our source tree, but if we did it would be pretty confusing if it was ignored...